crates/nu-cli/src/data/files.rs

```diff
@@ -2,6 +2,35 @@ use crate::prelude::*;
 use nu_errors::ShellError;
 use nu_protocol::{TaggedDictBuilder, UntaggedValue, Value};
 
+#[cfg(unix)]
+use std::os::unix::fs::FileTypeExt;
+
+fn get_file_type(md: &std::fs::Metadata) -> &str {
+    let ft = md.file_type();
+    let mut file_type = "Unknown";
```

Comment:
In general, if you can avoid mutability, it's a good idea to do so (simpler cognitive model). You can do that here by dropping this variable and writing your if statement this way:

```rust

if ft.is_dir() {
  "Dir"
} ...
} else {
  #[cfg(unix)]
  ...

  "Unknown"
}
```

Still going to land this PR as it is, but wanted to let you know about this 🙂

Reply:
Thanks, will remember for future, this is my first hands on in rust :)